filters/IterativeClosestPoint.hpp

```diff
@@ -53,8 +53,14 @@ class PDAL_DLL IterativeClosestPoint : public Filter
     double m_rotation_threshold;
     double m_translation_threshold;
     double m_mse_abs;
+    Arg *m_maxdistArg;
+    double m_maxdist;
+    Arg *m_matrixArg;
+    std::string m_matrixStr;
```

Comment:
I haven't looked at the Eigen code, but I would think that we could map the argument to Matrix4d or some such.  I think it has the necessary operators.  Or perhaps we should pull TransformationFilter::Transform out, rename and use that? Seems using eigen directly would be best, but a mapping also seems OK.

Reply:
I thought I'd run into issues trying to take Matrix4d directly via ProgramArgs in the past. And while I looked at TransformationFilter, it just seemed a bit overkill to copy/paste for this usage. Yes, if it had been generally available, I probably would've gone that route. Let me look again at using Eigen and I'll let you know where I get stuck.

Reply:
I wouldn't copy/paste. I'd just extract it into some private space, but better to use Eigen directly if that works. Ultimately, probably better to eigen-ize the transformation filter.